VMEncryption/main/handle.py

```diff
@@ -971,7 +1005,21 @@ def enable_encryption():
 
                     bek_util.store_bek_passphrase(encryption_config, extension_parameter.passphrase)
 
-                if extension_parameter.command == CommonVariables.EnableEncryptionFormatAll:
+                #If PrivatePreview.ConfidentialEncryptionTempDisk is set to True in public settings
+                #then temp disk encryption will happen for CVM type
+                  
+                public_settings = get_public_settings()
+                confidential_encryption_tempdisk = public_settings.get("PrivatePreview.ConfidentialEncryptionTempDisk")
+                encryptResourceDiskforCVM = False
+                if security_Type==CommonVariables.ConfidentialVM:
+                    if confidential_encryption_tempdisk:
+                        encryptResourceDiskforCVM = True
+                    else: 
+                        message = "Resource disk not encrypted. It is confidential VMs, but PrivatePreview.ConfidentialEncryptionTempDisk is not set"
+                        logger.log(msg=message)
+                                
+                if encryptResourceDiskforCVM or \
+                    extension_parameter.command == CommonVariables.EnableEncryptionFormatAll:
```

Comment:
What happens if user specifies EFA for CVM by mistake and did not set the preview flag? Will the temp disk be encrypted?